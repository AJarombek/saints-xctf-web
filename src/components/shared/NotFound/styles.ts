/**
 * JSS styles for the NotFound component.
 * @author Andrew Jarombek
 * @since 9/5/2020
 */

import {FontMixins} from "../../../styles/mixins";
import Colors from "../../../styles/colors";

export default {
    container: {
        display: 'flex',
        flexDirection: 'column',
        alignItems: 'center',
        justifyContent: 'center',
        minHeight: 'calc(100vh - 100px)',
        backgroundColor: Colors.lightBackground,
        margin: '100px 6% 0 6%',
    },
    logo: {
        height: 150,
        width: 150
    },
    title: {
        ...FontMixins.longway(),
        fontSize: 32
    },
    description: {
        ...FontMixins.robotoSlabThin(),
        fontSize: 18
    }
}
