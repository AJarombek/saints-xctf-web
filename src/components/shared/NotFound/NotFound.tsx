/**
 * Component for a not found page (a client-side 404 error page).
 * @author Andrew Jarombek
 * @since 9/5/2020
 */

import React from 'react';
import { createUseStyles } from 'react-jss';
import styles from './styles';
import classNames from 'classnames';
import { ClassValue } from 'classnames/types';

// @ts-ignore
import saintsXCTFLogo from '../../../../assets/saintsxctf_logo.png';

interface Props {
  fullPage: boolean;
  className?: ClassValue;
}

const useStyles = createUseStyles(styles);

const NotFound: React.FunctionComponent<Props> = ({ fullPage, className }) => {
  const classes = useStyles();

  return (
    <div className={classNames(classes.container, fullPage && classes.fullPageContainer, className)}>
      <img src={saintsXCTFLogo} alt="" className={classes.logo} />
      <h4 className={classes.title}>Page Not Found</h4>
      <p className={classes.description}>The requested page is invalid or no longer exists.</p>
    </div>
  );
};

export default NotFound;
