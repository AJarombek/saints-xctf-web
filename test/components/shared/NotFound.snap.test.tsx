/**
 * Snapshot test for the NotFound component.
 * @author Andrew Jarombek
 * @since 3/7/2021
 */

import React from 'react';
import renderer from 'react-test-renderer';
import NotFound from '../../../src/components/shared/NotFound';

describe('NotFound Snapshot Tests', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<NotFound fullPage={false} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
