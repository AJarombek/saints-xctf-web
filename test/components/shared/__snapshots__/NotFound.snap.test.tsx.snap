// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFound Snapshot Tests renders correctly 1`] = `
<div
  className="container-0-1-1"
>
  <img
    alt=""
    className="logo-0-1-3"
    src="test-file-stub"
  />
  <h4
    className="title-0-1-4"
  >
    Page Not Found
  </h4>
  <p
    className="description-0-1-5"
  >
    The requested page is invalid or no longer exists.
  </p>
</div>
`;
